src/main/java/org/spongepowered/common/mixin/core/entity/MixinEntityLivingBase.java

```diff
@@ -895,6 +907,59 @@ private void causeTrackDeathUpdate(EntityLivingBase entityLivingBase) {
         }
     }
 
+    private EquipmentSlotFabric inventory = new EquipmentSlotFabric(this);
+    private EnumMap<EntityEquipmentSlot, SlotLens> slotLens = new EnumMap<>(EntityEquipmentSlot.class);
+
+    @Inject(method = "onUpdate", locals = LocalCapture.CAPTURE_FAILHARD,
+            at = @At(value = "INVOKE", target = "Lnet/minecraft/entity/EntityLivingBase;getItemStackFromSlot(Lnet/minecraft/inventory/EntityEquipmentSlot;)Lnet/minecraft/item/ItemStack;"))
+    private void onGetItemStackFromSlot(CallbackInfo ci, int i, EntityEquipmentSlot[] slots, int j, int k,
+                                        EntityEquipmentSlot entityEquipmentSlot, ItemStack before) {
+        ItemStack after = this.getItemStackFromSlot(entityEquipmentSlot);
+        EntityLivingBase entity = EntityUtil.toNative((IMixinEntityLivingBase) this);
+        if (!ItemStack.areItemStacksEqual(after, before)) {
+            InventoryAdapter slotAdapter;
+            if (entity instanceof EntityPlayerMP) {
+                SlotLens slotLens;
+                IMixinInventoryPlayer inventory = (IMixinInventoryPlayer) ((EntityPlayerMP) entity).inventory;
+                PlayerInventoryLens inventoryLens = (PlayerInventoryLens) inventory.getRootLens();
+                switch (entityEquipmentSlot) {
+                case OFFHAND:
```

Comment:
Indentation

Reply:
@ImMorpheus Fixed in 5945012cb58ee83c4f235e90209ab464466d1ead